easybuild/easyblocks/e/elsi.py

```diff
@@ -0,0 +1,151 @@
+##
+# Copyright 2009-2019 Ghent University
+#
+# This file is part of EasyBuild,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://www.vscentrum.be),
+# Flemish Research Foundation (FWO) (http://www.fwo.be/en)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# https://github.com/easybuilders/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for ELSI, implemented as an easyblock
+
+@author: Miguel Dias Costa (National University of Singapore)
+"""
+import os
+import re
+from easybuild.easyblocks.generic.cmakemake import CMakeMake, setup_cmake_env
+from easybuild.framework.easyconfig import CUSTOM
+from easybuild.tools.build_log import EasyBuildError
+from easybuild.tools.modules import get_software_root, get_software_version
+
+
+class EB_ELSI(CMakeMake):
+    """Support for building ELSI."""
+
+    def __init__(self, *args, **kwargs):
+        """Initialize ELSI-specific variables."""
+        super(EB_ELSI, self).__init__(*args, **kwargs)
+        self.enable_sips = False
+        self.env_suff = '_MT' if self.toolchain.options.get('openmp', None) else ''
+
+    @staticmethod
+    def extra_options():
+        """Define custom easyconfig parameters for ELSI."""
+        extra_vars = {
+            'build_internal_pexsi': [True, "Build internal PEXSI solver", CUSTOM],
```

Comment:
I wonder if it would be more appropriate to have this as False or None.
Our default modus operandi is to rely on explicit dependencies, not internal ones.

Reply:
you're probably right... (but see above, https://github.com/easybuilders/easybuild-easyblocks/pull/1883#discussion_r357669460)

Reply:
I'm fine either way

Reply:
Yes, but that comment was about enabling pexsi, not about using the internal pexsi.

I'd say building_internal_pexsi should not be the default.
I.e. set it to None, and if there is no external pexsi, then use internal unless explicitly set to false in easyconfig file.
And complain if explicitly set to true with external pexsi dep defined.

Reply:
Ah, you're right, will do